server/routes/sockets/room.js

```diff
@@ -17,46 +66,160 @@ class Room {
+     */
+    kickUser(user) {
+        // Check if the user is in the list of connectedUsers.
+        const existingUser = this.getUserFromState(this.ioRef.sockets.sockets[user.socketID]);
+        if (existingUser) {
+            // Disconnect the user from the room channel.
+            this.ioRef.sockets.sockets[existingUser.socketID].leave(this.roomID);
+            // Remove the user from the list of connected users.
+            _.remove(this.roomState.connectedUsers, (connectedUser) => {
+                return user.socketID === connectedUser.socketID;
+            });
+            // Check if the voting phase can move to the results phase now
+            this.checkIfVotingComplete();
+            // Emit a room_state_changed message to everyone in the room.
+            this.emitRoomEvent('room_state_changed', { roomState: this.roomState });
+            // Emit a host_closed_connection event to the kicked user.
+            this.emitUserEvent('host_closed_connection', existingUser);
+        }
     }
 
     /**
      * Join a user to the room if they aren't already joined to the room.
-     * The user object consists of a username and a socket.
-     * Emit a user_joined event once they are connected.
+     * The user object consists of a nickname and a socket.
      */
-    joinUserToRoom(username, socket) {
-        // Check if the user is already in the list of connected users.
-        if (_.find(this.connectedUsers, (user) => {
-            return user.username === username;
-        })) {
+    joinUserToRoom(nickname, socket) {
+        // Check if the user's socket is already in the list of connected users.
+        const existingUser = this.getUserFromState(socket);
+        if (existingUser) {
             // User is already in the list of connected users.
-            return null;
+            this.emitUserEvent('user_already_in_room', existingUser);
         } else {
-            // Store info about the connectionin the connectedUsers array
-            this.connectedUsers.push({
-                username,
-                socket,
+            // Store info about the connection in the connectedUsers array
+            this.roomState.connectedUsers.push({
+                nickname,
+                socketID: socket.id,
             });
+            // Set the user's vote to null to signify they have not voted yet
+            this.roomState.voteByUserID[socket.id] = null;
             // Join the user's socket to the correct channel
             socket.join(this.roomID);
-            // Emit a user_joined event to everyone in the room
-            this.emitRoomEvent('user_joined', { username });
+            // Emit a room_state_changed event to everyone in the room
+            this.emitRoomEvent('room_state_changed', { roomState: this.roomState });
+            // Emit a join_success event to the user that just joined the room
+            this.emitUserEvent('join_success', socket, { roomState: this.roomState });
```

Comment:
The second parameter should be a user object, not a socket. See the Discord for the correct code.